streamer/src/nonblocking/quic.rs

```diff
@@ -137,7 +138,7 @@ async fn setup_connection(
 
         let remote_addr = connection.remote_address();
 
-        let (mut connection_table_l, stake) = {
+        let table_and_stake = {
```

Comment:
I think this entire code block should be its own function and it should return all the information necessary on how the connection should be added.

Reply:
Agree that it should be refactored into a separate function. As discussed on chat, we can probably defer it to its own PR.